src/Parser/RawBody.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class RawBody implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    protected $buffer = '';
+    protected $contentLength;
+
+    public function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
> My main issue with this ticket is that it tries to tackle multiple responsibilities. HTTP is a layered protocol and the newer specs make this much more obvious: Message framing (content-length, transfer-encoding etc.) should be handled before passing anything to the form parsers.

How do you suggest we proceed? I'm going to let the `FormParserFactory` send out a `411` in case of invalid request. But how do we communicate that back to the user? A way could be using promises